drv/lpc55-flash/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+    /// to also do that.
+    pub fn enable_interrupt_sources(&mut self) {
+        self.reg.int_set_enable.write(|w| {
+            w.fail().set_bit();
+            w.err().set_bit();
+            w.ecc_err().set_bit();
+            w.done().set_bit();
+            w
+        });
+    }
+
+    /// Turns off all interrupt sources in the flash controller (FAIL, ERR, ECC,
+    /// and DONE). This does _not_ clear any event that's pended in the NVIC
+    /// already, so you may still get an interrupt after doing this if you don't
+    /// clear pending.
+    pub fn disable_interrupt_sources(&mut self) {
+        self.reg.int_clr_enable.write(|w| {
+            w.fail().set_bit();
+            w.err().set_bit();
+            w.ecc_err().set_bit();
+            w.done().set_bit();
+            w
+        });
+    }
+
+    // API below this point is internal, but is exposed in case someone needs to
+    // do something weird, like attack the flash controller. The routines are
+    // deliberately documented less than the "official" API above.
+
+    pub fn issue_cmd(&mut self, cmd: FlashCmd) {
+        self.reg.cmd.write(|w| unsafe { w.cmd().bits(cmd as u32) });
+    }
+
+    pub fn set_single_word_number(&mut self, word_number: u32) {
+        // Technically, the write to STOPA this will incur is unnecessary.
+        // However, this avoids potentially undefined behavior if
+        // set_single_word_number is used before a command that actually takes a
+        // range. I'm being paranoid, and the cost is low.
+        self.set_word_range(word_number..=word_number);
+    }
+    
+    pub fn set_word_range(&mut self, range: RangeInclusive<u32>) {
+        self.reg.starta.write(|w| unsafe { w.starta().bits(*range.start()) });
+        // The code example in the user manual treats the STOPA word number for
+        // flash erase as _exclusive_ -- the code example is wrong. The
+        // documentation for the commands that take ranges correctly describes
+        // it as being inclusive. Otherwise how would you erase the last sector?
+        self.reg.stopa.write(|w| unsafe { w.stopa().bits(*range.end()) });
+    }
+
+    pub fn read_completion_status(&mut self) -> Option<Status> {
+        let s = self.reg.int_status.read();
+        if s.done().bit() {
+            Some(Status {
+                ecc_err: s.ecc_err().bit(),
+                err: s.err().bit(),
+                fail: s.fail().bit(),
+            })
+        } else {
+            None
+        }
+    }
+
+    pub fn clear_status_flags(&mut self) {
+        self.reg.int_clr_status.write(|w| {
+            w.done().set_bit();
+            w.ecc_err().set_bit();
+            w.err().set_bit();
+            w.fail().set_bit();
+            w
+        });
+    }
+}
+
+/// Raw status bits from the flash controller.
+///
+/// This type is exposed in case you want to do something weird with the raw
+/// flash controller operations; the higher-level APIs do not use it directly.
```

Comment:
Add a note here to the effect that a `Status` is only constructed when the `done` bit is set, so that bit isn't included?

Reply:
Good point, done!